src/mlpack/methods/kmeans/hamerly_kmeans_impl.hpp

```diff
@@ -59,70 +60,86 @@ double HamerlyKMeans<DistanceType, MatType>::Iterate(const arma::mat& centroids,
       ++distanceCalculations;
 
       // Update bounds, if this intra-cluster distance is smaller.
-      if (dist < minClusterDistances(i))
-        minClusterDistances(i) = dist;
-      if (dist < minClusterDistances(j))
-        minClusterDistances(j) = dist;
+      minClusterDistances(i) = std::min(minClusterDistances(i), dist);
+      minClusterDistances(j) = std::min(minClusterDistances(j), dist);
     }
   }
 
-  for (size_t i = 0; i < dataset.n_cols; ++i)
+  #pragma omp parallel
   {
-    const double m = std::max(minClusterDistances(assignments[i]),
-                              lowerBounds(i));
-
-    // First bound test.
-    if (upperBounds(i) <= m)
+    arma::mat threadNewCentroids(centroids.n_rows, centroids.n_cols,
+      arma::fill::zeros);
```

Comment:
```suggestion
    arma::mat threadNewCentroids(centroids.n_rows, centroids.n_cols);
```
(Armadillo defaults to `fill::zeros`, so we don't need to include it here.)